ceph/rados/pool_workflows.py

```diff
@@ -683,3 +683,230 @@ def fetch_pool_stats(self, pool: str) -> dict:
+                )
+                log.info(
+                    "Checking the ideal PG count on the pool and the threshold set. default is 3"
+                )
+                pool_data = self.rados_obj.get_pg_autoscale_status(pool_name=pool)
+                final_pg_num = pool_data["pg_num_final"]
+                log.debug(
+                    f"Pool : {pool} details as fetched from autoscale-status : {pool_data}"
+                )
+
+                if float(final_pg_num / init_pg_count) <= 3.0:
+                    log.info(
+                        "PG count not expected to increase as the final pg num is less than the threshold"
+                    )
+                    if modify_threshold:
+                        log.info(
+                            "Param passed to modify the threshold set. Changing it to 1.3 from 3."
+                        )
+                        self.modify_autoscale_threshold(threshold=1.3)
+                        time.sleep(60)
+                    else:
+                        no_count_change = True
+
+            if not no_count_change:
+                log.debug(
+                    f"PG count on pool {pool} post addition of bulk flag : {pg_count_bulk_true}"
+                    f"Starting to wait for PG count on the pool to go from {init_pg_count} to"
+                    f" {pg_count_bulk_true} while checking for PG inactivity"
+                )
+
+                inactive_pg = 0
+                endtime = datetime.datetime.now() + datetime.timedelta(seconds=14000)
+                while datetime.datetime.now() < endtime:
+                    pool_pg_num = self.rados_obj.get_pool_property(
+                        pool=pool, props="pg_num"
+                    )["pg_num"]
+                    if pool_pg_num == pg_count_bulk_true:
+                        log.info(
+                            f"PG count on pool {pool} is achieved post adding the bulk flag"
+                        )
+                        break
+                    log.info(
+                        f"PG count on pool {pool} has not reached desired levels."
+                        f"Expected : {pg_count_bulk_true}, Current : {pool_pg_num}"
+                    )
+                    if not self.rados_obj.check_inactive_pgs_on_pool(pool_name=pool):
+                        log.error(f"Inactive PGs found on pool : {pool}")
+                        inactive_pg += 1
+
+                    log.info(
+                        "Sleeping for 60 secs and checking the PG states and PG count"
+                    )
+                    time.sleep(60)
+                else:
+                    log.error(
+                        f"pg_num on pool {pool} did not reach the desired levels of PG count "
+                        f"with bulk flag enabled"
+                        f"Expected : {pg_count_bulk_true}"
+                    )
```

Comment:
Please check that the function returns without resetting the default threshold value which was assigned 1.3 in the previous steps.

Reply:
fixed it. reverting back in finally in case of failures.